src/base/static/js/views/geometry-editor-view.js

```diff
@@ -0,0 +1,583 @@
+    this.editingLayerGroup = new L.FeatureGroup();
+    this.DRAWING_DEFAULTS = {
+      color: "#f06eaa",
+      opacity: 0.5,
+      fillColor: "#f06eaa",
+      fillOpacity: 0.2,
+      fill: {
+        color: "fillColor",
+        opacity: "fillOpacity"
+      },
+      stroke: {
+        color: "color",
+        opacity: "opacity"
+      }
+    };
+    this.colorpickerSettings = {
+      editMode: "fill",
+      color: this.DRAWING_DEFAULTS.color,
+      opacity: this.DRAWING_DEFAULTS.opacity,
+      fillColor: this.DRAWING_DEFAULTS.fillColor,
+      fillOpacity: this.DRAWING_DEFAULTS.fillOpacity
+    };
+
+    // Init the colorpicker
+    $("body").append("<div class='toolbar-colorpicker-container'></div>");
+    $(".toolbar-colorpicker-container").spectrum({
+      flat: true,
+      showButtons: false,
+      showInput: true,
+      showAlpha: true,
+      
+      // Convert to rgba() format
+      color: tinycolor(self.colorpickerSettings.fillColor)
+        .setAlpha(self.colorpickerSettings.fillOpacity)
+        .toRgbString(),
+      move: function(color) {
+        if (self.placeDetailView) {
+          self.placeDetailView.onModified();
+        }
+        if (self.editingLayerGroup.getLayers().length > 0) {
+          if (self.colorpickerSettings.editMode === "fill") {
+            self.editingLayerGroup.getLayers()[0].setStyle({
+              fillColor: color.toHexString(),
+              fillOpacity: color.getAlpha()
+            });
+            self.colorpickerSettings.fillColor = color.toHexString();
+            self.colorpickerSettings.fillOpacity = color.getAlpha();
+          } else if (self.colorpickerSettings.editMode === "stroke") {
+            self.editingLayerGroup.getLayers()[0].setStyle({
+              color: color.toHexString(),
+              opacity: color.getAlpha()
+            });
+            self.colorpickerSettings.color = color.toHexString();
+            self.colorpickerSettings.opacity = color.getAlpha();
+          }
+        }
+      },
+      change: function() {
+        //console.log("color change");
+      }
+    });
+
+    $(".sp-picker-container")
+      .addClass("hidden")
+      .prepend(Handlebars.templates["colorpicker-controls"]());
+
+    $(".sp-choose").on("click", function(evt) {
+      self.onClickColorEditModeChange(evt);
+    });
+
+    var generateGeometry = function(layer) {
+      var buildCoords = function(layer) {
+        var coordinates = [],
+        latLngs = layer.getLatLngs();
+        for (var i = 0; i < latLngs.length; i++) {
+          coordinates.push([latLngs[i].lng, latLngs[i].lat]);
+        }
+
+        return coordinates;
+      };
+
+      if (self.geometryType === "polygon" ||
+          self.geometryType === "rectangle" ||
+          self.geometryType === "Polygon") {
+        
+        var coordinates = buildCoords(layer),
+        latLngs = layer.getLatLngs();
+        
+        // Make sure the final polygon vertex exactly matches the first. The
```

Comment:
If I understand the way the PostGIS plugin works, the final vertex a list of polygon vertices needs to exactly match the first in order to have valid polygon geometry. So the duplication is intentional as it's required by the database.